azuredevops/internal/service/serviceendpoint/commons.go

```diff
@@ -28,34 +28,30 @@ func genBaseServiceEndpointResource(f flatFunc, e expandFunc) *schema.Resource {
 		Update:   genServiceEndpointUpdateFunc(f, e),
 		Delete:   genServiceEndpointDeleteFunc(e),
 		Importer: tfhelper.ImportProjectQualifiedResourceUUID(),
-		Schema:   genBaseSchema(),
-	}
-}
-
-func genBaseSchema() map[string]*schema.Schema {
-	return map[string]*schema.Schema{
-		"project_id": {
-			Type:     schema.TypeString,
-			Required: true,
-			ForceNew: true,
-		},
-		"service_endpoint_name": {
-			Type:         schema.TypeString,
-			Required:     true,
-			ValidateFunc: validate.NoEmptyStrings,
-			ForceNew:     true,
-		},
-		"description": {
-			Type:     schema.TypeString,
-			Optional: true,
-			Default:  "Managed by Terraform",
-		},
-		"authorization": {
-			Type:     schema.TypeMap,
-			Optional: true,
-			Computed: true,
-			Elem: &schema.Schema{
-				Type: schema.TypeString,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_endpoint_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+				ForceNew:     true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Managed by Terraform",
+			},
+			"authorization": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
```

Comment:
Right now I'm not 100% sure if you can add validation on map elements. But if so add `ValidateFunc: validate.NoEmptyStrings`

Reply:
I don't think they work

Reply:
But I added one anyways!